source/agora/node/Ledger.d

```diff
@@ -132,6 +132,12 @@ public class Ledger
 
     public bool acceptTransaction (Transaction tx) @safe
     {
+        if (!this.pool.isValidTransaction(tx))
```

Comment:
Could you move this check into the `isValidTransaction` method in the `Ledger`? That way we have a single method where we check transaction validation.